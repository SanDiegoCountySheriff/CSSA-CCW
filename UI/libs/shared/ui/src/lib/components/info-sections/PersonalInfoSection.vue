<template>
  <v-container class="info-section-container rounded">
    <v-card
      :color="props.color"
      class="rounded"
    >
      <v-card-title class="ml-3">
        {{ $t('Personal Information') }}
      </v-card-title>
      <v-card-text class="m-5">
        <v-row>
          <v-col>
            <div class="info-row">
              <h3>{{ $t('Last name:') }}</h3>
              <p class="info-text">
                {{ props.personalInfo.lastName }}
              </p>
            </div>
          </v-col>
          <v-col>
            <div class="info-row">
              <h3>{{ $t('First name:') }}</h3>
              <p class="info-text">
                {{ props.personalInfo.firstName }}
              </p>
            </div>
          </v-col>
        </v-row>
        <v-row>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('Middle name: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.middleName }}
              </p>
            </div>
          </v-col>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('No middle name: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.noMiddleName }}
              </p>
            </div>
          </v-col>
        </v-row>
        <v-row>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('Maiden name: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.maidenName }}
              </p>
            </div>
          </v-col>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('Suffix name: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.suffix }}
              </p>
            </div>
          </v-col>
        </v-row>
        <v-row>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('Social Security number: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.ssn }}
              </p>
            </div>
          </v-col>
          <v-col>
            <div class="info-row">
              <h3>
                {{ $t('Marital status: ') }}
              </h3>
              <p class="info-text">
                {{ props.personalInfo.maritalStatus }}
              </p>
            </div>
          </v-col>
        </v-row>
      </v-card-text>
    </v-card>
  </v-container>
</template>

<script setup lang="ts">
/**
 * This Should be able to be used in both the public and the admin.
 * to change the size put this into a container in the parent component and change
 * the size of the container
 * Also the bg color can be changed in the props.
 */
import { PersonalInfoType } from '@shared-ui/types/defaultTypes';

interface personalInfoSectionProps {
  personalInfo: PersonalInfoType;
  color: string;
}
const props = defineProps<personalInfoSectionProps>();
</script>

<style lang="scss">
.info-section-container {
  width: 100%;
  margin: 0;
  padding: 0;
}
.info-row {
  display: flex;
  flex-direction: row;
  margin-left: 0.5rem;
}

.info-text {
  margin-left: 0.5rem;
}
</style>
