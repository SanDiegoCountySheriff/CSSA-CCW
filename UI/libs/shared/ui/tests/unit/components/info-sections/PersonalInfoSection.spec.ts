import PersonalInfoSection from '@shared-ui/components/info-sections/PersonalInfoSection.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('PersonalInfoSection', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    //@ts-ignore
    wrapper = mount(PersonalInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => {
          text;
        },
      },
      propsData: {
        color: 'teal lighten-5',
        personalInfo: {
          lastName: 'last',
          firstName: 'first',
          middleName: 'middle',
          noMiddleName: false,
          maidenName: '',
          suffix: '',
          ssn: 'ssn',
          maritalStatus: 'single',
        },
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
