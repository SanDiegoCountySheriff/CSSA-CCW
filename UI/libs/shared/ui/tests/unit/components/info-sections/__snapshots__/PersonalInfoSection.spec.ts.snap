// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInfoSection should match the snapshot 1`] = `
<div class="container info-section-container rounded">
  <transition-stub name="expand-transition" class="font-weight-bold text-xl text-left mb-5">
    <div class="v-banner v-sheet theme--light v-banner--is-mobile">
      <div class="v-banner__wrapper">
        <div class="v-banner__content">
          <div class="v-banner__text">

          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-4" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-7" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-10" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-13" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-16" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-19" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined">
        <div class="v-input__control">
          <div class="v-input__slot">
            <fieldset aria-hidden="true">
              <legend style="width: 0px;"><span class="notranslate">​</span></legend>
            </fieldset>
            <div class="v-text-field__slot"><input id="input-22" readonly="readonly" type="text"></div>
          </div>
          <div class="v-text-field__details">
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
